### PR TITLE
perf(runtime): optimize Deno.file open & stream

### PR DESCRIPTION
**Pull request by littledivy**
Opening and closing files is 2.5x faster
```
# Deno new
open_file_sync     7.55 µs/iter     (7.51 µs … 7.72 µs)   7.56 µs   7.72 µs   7.72 µs
# Deno `main`
open_file_sync    18.31 µs/iter   (17.79 µs … 39.75 µs)  18.25 µs  19.62 µs  20.62 µs
```

ext/flash send_file benchmark now gives 7GB/s transfer rate. 1.3x faster than `main`:
```
Transfer/sec:      5.79GB # main
Transfer/sec:      7.56GB # new
```

### CODE REVIEW

File: ext/web/06_streams.js [hunk 1]
```diff
@@ -658,7 +658,9 @@
    * @returns {ReadableStream<Uint8Array>}
    */
   function readableStreamForRid(rid, unrefCallback) {
-    const stream = new ReadableStream({
```
Comment (littledivy) [inline ext/web/06_streams.js:661]: something to note: ReadableStream initialization triggers a pretty serious deopt in V8. From a flamegraph of the send file server on `main`:

image

**Comment by aapoalas** [comment]
> Opening and closing files is 2.5x faster
> 
> ```
> # Deno `main`
> open_file_sync     7.55 µs/iter     (7.51 µs … 7.72 µs)   7.56 µs   7.72 µs   7.72 µs
> # Deno new
> open_file_sync    18.31 µs/iter   (17.79 µs … 39.75 µs)  18.25 µs  19.62 µs  20.62 µs
> ```

Am I reading this wrong somehow? I read this as the new `open_file_sync` taking 18 microseconds and the old one taking only 7.5 microseconds, ie. old one being faster.

**Comment by littledivy** [comment]
> Am I reading this wrong somehow? I read this as the new open_file_sync taking 18 microseconds and the old one taking only 7.5 microseconds, ie. old one being faster.

Oopsie, wrong labels.

**Review by bartlomieju — APPROVED** [review]
LGTM

**Pull request merged**